cmd/commands.go

```diff
@@ -0,0 +1,189 @@
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	buildCommand.Flags().StringArray("with", []string{}, "caddy modules package path to include in the build")
+	buildCommand.Flags().String("output", "", "change the output file name")
+	buildCommand.Flags().StringArray("replace", []string{}, "like --with but for Go modules")
+	buildCommand.Flags().StringArray("embed", []string{}, "embeds directories into the built Caddy executable to use with the `embedded` file-system")
+}
+
+var versionCommand = &cobra.Command{
+	Use:   "version",
+	Short: "Prints xcaddy version",
+	RunE: func(cm *cobra.Command, args []string) error {
+		fmt.Println(xcaddyVersion())
+		return nil
+	},
+}
+
+var buildCommand = &cobra.Command{
+	Use: `build [<caddy_version>]
+    [--output <file>]
+    [--with <module[@version][=replacement]>...]
+    [--replace <module[@version]=replacement>...]
+    [--embed <[alias]:path/to/dir>...]`,
+	Long: `
+<caddy_version> is the core Caddy version to build; defaults to CADDY_VERSION env variable or latest.
+This can be the keyword latest, which will use the latest stable tag, or any git ref such as:
+
+A tag like v2.0.1
+A branch like master
+A commit like a58f240d3ecbb59285303746406cab50217f8d24
+
+Flags: 
+ --output changes the output file.
+
+ --with can be used multiple times to add plugins by specifying the Go module name and optionally its version, similar to go get. Module name is required, but specific version and/or local replacement are optional.
+
+ --replace is like --with, but does not add a blank import to the code; it only writes a replace directive to go.mod, which is useful when developing on Caddy's dependencies (ones that are not Caddy modules). Try this if you got an error when using --with, like cannot find module providing package.
+
+ --embed can be used multiple times to embed directories into the built Caddy executable. The directory can be prefixed with a custom alias and a colon : to use it with the root directive and sub-directive. Example: xcaddy build --embed foo:./sites/foo --embed bar:./sites/bar (This allows you to serve 2 sites from 2 different embedded directories, which are referenced by aliases, from a single Caddy executable).
+`,
+	Short: "Compile custom caddy binaries",
+	Args:  cobra.MaximumNArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var output string
+		var plugins []xcaddy.Dependency
+		var replacements []xcaddy.Replace
+		var embedDir []string
+		var argCaddyVersion string
+		if len(args) > 0 {
+			argCaddyVersion = args[0]
+		}
+		withArgs, err := cmd.Flags().GetStringArray("with")
+		if err != nil {
+			return fmt.Errorf("unable to parse --with arguments: %s", err.Error())
+		}
+
+		replaceArgs, err := cmd.Flags().GetStringArray("replace")
+		if err != nil {
+			return fmt.Errorf("unable to parse --replace arguments: %s", err.Error())
+		}
+		for _, withArg := range withArgs {
+			mod, ver, _, err := splitWith(withArg)
+			if err != nil {
+				return err
+			}
```

Comment:
@armadi1809 The third returned value here, which was called `repl` in earlier [code](https://github.com/caddyserver/xcaddy/blob/ff67f855c7a41a07d95849d90372aee939456fb4/cmd/main.go#L83-L108), is ignored. It shouldn't be ignored. It shouldn't be.

Can you copy the older logic for the fix? If you're too busy, I can do it.